packages/documentation/docs/SAG/SAG-for-starknet.md

```diff
@@ -1,6 +1,135 @@
 ---
 id: SAG-starknet
 title: "SAG for Starknet"
---- 
+---
+
+# SAG for Starknet
+
+The **SAG for Starknet** implementation enables verification of Spontaneous Anonymous Group (SAG) signatures on the Starknet platform. This implementation leverages Starknet-specific optimizations to perform ring signature verification efficiently on-chain, supporting both SECP256K1 and ED25519 elliptic curves.
+
+The SAG verification functionality on Starknet is provided by the `starknet-sag-ts` package from [Alice's Ring](https://github.com/Cypher-Laboratory/Alice-s-Ring-SAG-TS), a TypeScript library for creating SAG signatures.
+
+## Key Differences in the Starknet Implementation
+
+The Starknet version of the SAG verifier shares the same fundamental logic as the EVM version, with a few important adjustments to optimize for the Starknet ecosystem:
+
+1. **Poseidon Hash Function**: The signer and verifier use the Poseidon hashing function, which is computationally cheaper on Starknet compared to traditional hashing algorithms.
+
+2. **Garaga for Multi-Scalar Multiplication (MSM)**: The implementation is using Garaga allowing efficient on-chain multi-scalar multiplication (MSM). The process incorporates hints passed alongside the ring signature to perform MSM efficiently.
+
+## Generating a Ring Signature for Starknet
+
+Here's how to generate a ring signature for starknet using Alice's Ring in TypeScript:
+
+```typescript
+import {
+  RingSignature,
+  Curve,
+  CurveName,
+  Point,
+} from "@cypher-laboratory/alicesring-sag-starknet";
+
+const curve = new Curve(CurveName.SECP256K1);
+const ring: Point[] = []; // Your ring of public keys
+const message = "Hello, Alice's Ring!";
+
+const signerPrivateKey = BigInt("0x..."); // The signer's private key
+async function ringSignForStarknet() {
+  // Sign
+  const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);
+
+  // Verify locally
+  console.log("Is signature valid?", signature.verify()); // Should output: true
+
+  const signature = RingSignature.sign(ring, signerPrivKey, message, curve);
+  const rawCallData = await signature.getCallData();
+  console.log(rawCallData);
+}
+ringSignForStarknet(); 
```

Comment:
```suggestion

// Sign
const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);

// Verify locally
console.log("Is signature valid?", signature.verify()); // Should output: true

(async () => {
  // Get the signature data to send to the contract
  const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);
  const rawCallData = await signature.getCallData();
  console.log(rawCallData);
})();
```

- snippet was not valid (signature was defined twice in the same block and `signerPrivKey` was undefined)
- replaced the function by a simple async block (increase readability)
- only let the async call in the async block
- explained what is done in the `rawCallData` generation 

Can you also explain a little more the link between `rawCallData` and the contract (how is it used, if it is a function argument, explain whch one, ...). Maybe add a snippet where you generate a signature and verify it using the contract